### DISCUSSION TITLE
if using '.lean()' field 'id' is not in the document type.

### DISCUSSION THREAD

**Discussion by Manfredik** [typegoose/typegoose#610] [Q&A]
when using `.lean()` field `id` is not in the document type, you have to use crutches and typecasting, which is not very good

```ts
export class Landing {
	@prop()
	id!: number;

	@prop()
	type!: number;
.....
}

const landings: Pick<Landing, 'id' | 'type'> = await LandingModel.find({}).select({id: 1, type: 1}).lean();
```

`Property 'id' is missing in type 'DocumentDefinition<DocumentType<Landing>>' but required in type 'Pick<Landing, "id" | "type">'. `

**Comment by hasezoey** [comment]
This is expected behavior, because `id` in not part of the mongoose types for the document, so you would have to define them yourself (see the [FAQ](https://typegoose.github.io/typegoose/docs/guides/faq#i-want-to-the-return-document-with-property-id-instead-of-_id))

Also, when using `.lean`, it returns the raw MongoDB response (not being cast to a `Document`), so it does not include virtuals, if you need virtuals included, you would have to remove `.lean` and after the query use `.toJSON({ virtuals: true })` (or just `.toJSON()` with the schemaOptions set), see [this stackoverflow](https://stackoverflow.com/questions/18363472/mongoose-lean-query-virtuals-not-showing)
